pkg/kvcache/kvblock/instrumented_index.go

```diff
@@ -55,6 +55,29 @@ func (m *instrumentedIndex) Lookup(
 	metrics.LookupRequests.Inc()
 
 	pods, err := m.next.Lookup(ctx, keys, podIdentifierSet)
+	go recordHitMetrics(pods)
 
 	return pods, err
 }
+
+func recordHitMetrics(keyToPods map[Key][]string) {
```

Comment:
For a follow-up PR (will open an issue based on this comment) - wonder if this better fits to be recorded by the `kvblock.Scorer` and not the index.